### PR TITLE
Merge crowdsec lapi

### PR DESCRIPTION
**Pull request by AlteredCoder**

### CODE REVIEW

File: pkg/leakybucket/bucket.go [hunk 1]
```diff
@@ -256,7 +259,10 @@ func LeakRoutine(leaky *Leaky) {
 			if leaky.timedOverflow {
 				BucketsOverflow.With(prometheus.Labels{"name": leaky.Name}).Inc()
 
-				alert = NewAlert(leaky, ofw)
+				alert, err := NewAlert(leaky, ofw)
```
Comment (registergoofy) [inline pkg/leakybucket/bucket.go:262]: there's a scope issue here
alert is not the same as the alert out of the if check.

**Review by registergoofy — CHANGES_REQUESTED** [review]
minor things.
I think this require more reviews

File: wizard.sh [hunk 2]
```diff
@@ -435,7 +449,7 @@ main() {
         log_info "installing crowdsec"
         install_crowdsec
          # api register
-        ${CSCLI_BIN_INSTALLED} api register >> /etc/crowdsec/config/api.yaml || ${CSCLI_BIN_INSTALLED} api reset >> /etc/crowdsec/config/api.yaml || log_err "unable to register, skipping crowdsec api registration"
+        ${CSCLI_BIN_INSTALLED} watchers add --machine "$(cat /etc/machine-id)" --password "tmp^Crowdsec^password0.3.2"
```
Comment (buixor) [inline wizard.sh:452]: I see what you did there xD

**Review by buixor — CHANGES_REQUESTED** [review]
great wip

File: pkg/leakybucket/bucket.go (hunk 1 above)
Comment (buixor) [reply pkg/leakybucket/bucket.go]: @registergoofy can you be more explicit ?

File: cmd/crowdsec-cli/watchers.go [hunk 3]
```diff
@@ -167,11 +168,46 @@ The watcher will be validated automatically.
...
+				} else if csConfig.API.Server != nil {
+					apiURL = csConfig.API.Server.ListenURI
+				}
+				if apiURL == "" {
+					log.Fatalf("unable to dump an api URL. Please provide it in your configuration or with the -u parameter")
+				}
+			}
+
+			apiCfg := csconfig.ApiCredentialsCfg{
+				Login:    machineID,
```
[the remark this answers is not in the document]
Comment (AlteredCoder) [reply cmd/crowdsec-cli/watchers.go:190]: Maybe we can do this task in a later merge request. We must don't forget it

File: cmd/crowdsec/crowdsec.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			err := runParse(inputLineChan, inputEventChan, *parsers.ctx, parsers.nodes)
+			if err != nil {
+				log.Errorf("runParse error : %s", err)
+				return err
+			}
+			return nil
+		})
+	}
+
+	bucketsTomb.Go(func() error {
```
[the remark this answers is not in the document]
Comment (AlteredCoder) [reply cmd/crowdsec/crowdsec.go:57]: Please not that for a futur MR.

File: pkg/apiserver/controllers/alerts.go [hunk 5]
```diff
@@ -85,12 +87,13 @@ func FormatAlerts(result []*ent.Alert) models.AddAlertsRequest {
...
 }
 
+// CreateAlert : write received alerts in body to the database
 func (c *Controller) CreateAlert(gctx *gin.Context) {
 	var input models.AddAlertsRequest
 
 	claims := jwt.ExtractClaims(gctx)
 	/*TBD : use defines rather than hardcoded key to find back owner*/
-	machineId := claims["id"].(string)
+	machineID := claims["id"].(string)
```
[the remark this answers is not in the document]
Comment (AlteredCoder) [reply pkg/apiserver/controllers/alerts.go:96]: please not that for a future MR

**Pull request merged**